src/helm/proxy/clients/huggingface_model_registry.py

```diff
@@ -0,0 +1,94 @@
+from typing import Dict, Optional
+from dataclasses import dataclass
+import re
+from helm.common.hierarchical_logger import hlog
+from helm.proxy.models import (
+    Model,
+    MODEL_NAME_TO_MODEL,
+    TEXT_MODEL_TAG,
+    FULL_FUNCTIONALITY_TEXT_MODEL_TAG,
+)
+
+
+@dataclass(frozen=True)
+class HuggingFaceModelConfig:
+    namespace: Optional[str]
+    """Name of the group or user that owns the model. e.g. 'stanford-crfm'
+
+    May be None if the model (e.g. gpt2) does not have a namespace."""
+
+    model_name: str
+    """Name of the model. e.g. 'BioMedLM'
+
+    Does not include the namespace. ."""
+
+    revision: Optional[str]
+    """Revision of the model to use e.g. 'main'.
+
+    If None, use the default revision."""
+
+    @property
+    def model_id(self) -> str:
+        """Return the model ID.
+
+        Examples:
+        - 'gpt2'
+        - 'stanford-crfm/BioMedLM'"""
+        if self.namespace:
+            return f"{self.namespace}/{self.model_name}"
+        return self.model_name
+
+    @staticmethod
+    def from_string(raw: str) -> "HuggingFaceModelConfig":
+        """Parses a string in the format "[namespace/]model_name[#revision]" to a HuggingFaceModelConfig.
+
+        Examples:
+        - 'gpt2'
+        - 'stanford-crfm/BioMedLM'
+        - 'stanford-crfm/BioMedLM#main'"""
+        pattern = r"((?P<namespace>[^/#]+)/)?(?P<model_name>[^/#]+)(#(?P<revision>[^/#]+))?"
+        match = re.fullmatch(pattern, raw)
+        if not match:
+            raise ValueError(f"Could not parse model name: '{raw}'; Expected format: [namespace/]model_name[#revision]")
+        model_name = match.group("model_name")
+        assert model_name
+        return HuggingFaceModelConfig(
+            namespace=match.group("namespace"), model_name=model_name, revision=match.group("revision")
+        )
+
+
+_huggingface_model_registry: Dict[str, HuggingFaceModelConfig] = {}
+
+
+def register_huggingface_model_config(raw_model_config: str) -> HuggingFaceModelConfig:
+    """Parses a string in the format '[namespace/]model_name[#revision]'"""
```

Comment:
comment seems wrong? doesn't mention the whole registration thing, but maybe that's idiomatic (docstring makes me thing it's just an alias for HuggingFaceModelConfig.from_string

Reply:
Fixed comment.